packages/amplify_auth_cognito/ios/Classes/FlutterSignUpRequest.swift

```diff
@@ -13,41 +13,51 @@
  * express or implied. See the License for the specific language governing
  * permissions and limitations under the License.
  */
- 
+
 import Foundation
 import Amplify
 import AmplifyPlugins
 import amplify_core
 
 struct FlutterSignUpRequest {
-  var username: String
-  var password: String
-  var userAttributes: [AuthUserAttribute] = []
-  var options: Dictionary<String, Any>? = [:]
-  init(dict: NSMutableDictionary){
-    self.username = dict["username"] as! String
-    self.password = dict["password"] as! String
-    self.options = dict["options"] as? Dictionary<String, Any>
-    self.userAttributes = self.createAuthUserAttributes(options: dict["options"] as! Dictionary<String, Any>)
-  }
+    var username: String
+    var password: String
+    var options: AuthSignUpRequest.Options?
+    init(dict: NSMutableDictionary){
+        self.username = dict["username"] as! String
+        self.password = dict["password"] as! String
+        self.options = createOptions(options: dict["options"] as? Dictionary<String, Any>)
+    }
+    
+    private func createOptions(options: Dictionary<String, Any>?) -> AuthSignUpRequest.Options? {
+        if (options == nil) {
+            return nil
+        }
+        var formattedAttributes: Array<AuthUserAttribute> = Array()
+        if (options!["userAttributes"] is Dictionary<String, String>) {
+            let rawAttributes: Dictionary<String, Any> = options!["userAttributes"] as! Dictionary<String, String>
+            for attr in rawAttributes {
+                formattedAttributes.append(createAuthUserAttribute(key: attr.key, value: attr.value as! String))
+            }
+        }
+        return AuthSignUpRequest.Options(userAttributes: formattedAttributes)
+        
+    }
```

Comment:
Is it possible for the initial guard clause to just be
```swift
if (!(options?["userAttributes"] is Dictionary<String, String>)) {
    return nil
}
// rest of function without if condition
```
It would reduce some of the nesting and improve readability if so. I think it'd also make it possible to just assign a mapped array directly to `formattedAttributes` similar to my kotlin suggestion above.

Reply:
You are right. I am going to make this update. I think I was mimicking what I had done on the Android side. However, on Android the nested if statements has a purpose since `options` can contain userAttributes and/or validationData. On iOS we aren't doing anything with the validationData. That seems like it might be a bug.

@haverchuck It doesn't seem like there is any way to include "validationData" in the sign up request in amplify-ios. Amplify-flutter seems to just be ignoring that value on iOS. On Android validationData is part of the `AWSCognitoAuthSignUpOptions` that gets included in the request. Is that a known platform discrepancy?

Reply:
Yeah - this is something we need to fix (part of the clientmetadata work that Hui and myself have implemented piecemeal).